integration/spark/src/main/common/java/io/openlineage/spark/agent/transformers/SqlDWRelationTransformer.java

```diff
@@ -0,0 +1,86 @@
+package io.openlineage.spark.agent.transformers;
+
+import io.openlineage.spark.agent.lifecycle.plan.DatasetSource;
+import java.io.ByteArrayInputStream;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.IllegalClassFormatException;
+import java.security.ProtectionDomain;
+import java.util.Arrays;
+import javassist.ClassPool;
+import javassist.CtClass;
+import javassist.CtMethod;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link ClassFileTransformer} that rewrites both the {@link
+ * com.google.cloud.spark.bigquery.BigQueryRelation} class as well as the v2 {@link
+ * com.google.cloud.spark.bigquery.v2.BigQueryDataSourceReader} class to implement the {@link
+ * DatasetSource} interface by introducing the {@link DatasetSource#name()} method, returning the
+ * fully qualified table name (&lt;projectId&gt;.&lt;dataset&gt;.&lt;table&gt;), and the {@link
+ * DatasetSource#namespace()} method, returning the literal string "bigquery".
+ */
+@Slf4j
+public class SqlDWRelationTransformer implements ClassFileTransformer {
```

Comment:
Sorry, I should have mentioned that this isn't necessary. The agent approach is going away and isn't advertised anymore. We can skip merging this code.

Reply:
I will drop it from the PR!  Thank you!